vaadin-markdown-flow-parent/vaadin-markdown-flow/src/test/java/com/vaadin/flow/component/markdown/tests/MarkdownTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2000-2025 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.markdown.tests;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.internal.PendingJavaScriptInvocation;
+import com.vaadin.flow.component.markdown.Markdown;
+import com.vaadin.flow.server.VaadinSession;
+
+public class MarkdownTest {
+
+    private final UI ui = new UI();
+    private Markdown markdown;
+
+    @Before
+    public void setup() {
+        markdown = new Markdown();
+
+        var mockSession = Mockito.mock(VaadinSession.class);
+        ui.getInternals().setSession(mockSession);
+        ui.add(markdown);
+    }
+
+    @Test
+    public void testInitialMarkdownContent() {
+        Assert.assertNull(markdown.getContent());
+    }
+
+    @Test
+    public void testSetContent() {
+        markdown.setContent("**Hello** _World_");
+        assertUpdateMarkdownCall(markdown, "**Hello** _World_", false);
+    }
+
+    @Test
+    public void testGetMarkdown() {
+        markdown.setContent("**Hello** _World_");
+        Assert.assertEquals("**Hello** _World_", markdown.getContent());
+    }
+
+    @Test
+    public void overloadedConstructor_testGetMarkdown() {
+        markdown = new Markdown("**Hello** _World_");
+        Assert.assertEquals("**Hello** _World_", markdown.getContent());
+    }
+
+    @Test
+    public void testAppendContent() {
```

Comment:
Let's also add assertions that serverContent was updated immediately, and getContent returns the correct results. 

The tests below that use setContent multiple times could also have similar assertions for completeness sake.

Reply:
Done